src/App.tsx

```diff
@@ -1,19 +1,138 @@
-import React from 'react';
+/* eslint-disable import/no-cycle */
+import React, { useState } from 'react';
+
 import './App.css';
 
+import todos from './api/todos';
 import users from './api/users';
 
-const App: React.FC = () => {
+import { User } from './components/UserInfo/UserInfo';
+import { Todo } from './components/TodoInfo/TodoInfo';
+import { TodoList } from './components/TodoList/TodoList';
+
+export interface PreparedTodo extends Todo {
+  user: User | null;
+}
+
+export const preparedTodos: PreparedTodo[] = todos.map((todo) => ({
+  ...todo,
+  user: users.find((user) => (
+    user.id === todo.userId
+  )) || null,
+}) as PreparedTodo);
+
+export const App: React.FC = () => {
+  const [todo, setTodos] = useState([...preparedTodos]);
+  const [title, setTitle] = useState('');
+  const [titleDirty, setTitleDirty] = useState(false);
+  const [userName, setUserName] = useState('Choose a user');
+  const [userNameDirty, setUserNameDirty] = useState(false);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleDirty(false);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    setUserName(value);
+    setUserNameDirty(false);
+  };
+
+  const addTodo = () => {
+    const newUser = users.find(
+      user => user.name === userName,
+    ) || null;
+
+    let userId = 0;
+
+    if (newUser) {
+      userId = newUser.id;
+    }
+
+    const newTodo = {
+      id: Math.max(...todos.map(t => t.id)) + 1,
```

Comment:
Same id's here, use your `todo` state
<img width="501" alt="image" src="https://user-images.githubusercontent.com/37161315/180026902-a93232d4-436f-4511-9da3-36018d292790.png">
```suggestion
      id: Math.max(...todo.map(t => t.id)) + 1,
```